### DISCUSSION TITLE
What does it mean to both append and prepend a value to a variable?

### DISCUSSION THREAD

**Discussion by bound-variable** [fish-shell/fish-shell#8523] [Q&A] [answered]
Per the [fish shell documentation](https://fishshell.com/docs/current/cmds/set.html):

> **append**: causes the values to be appended to the current set of values for the variable. This can be used with --prepend to both append and prepend at the same time.
> 
> **prepend**: causes the values to be prepended to the current set of values for the variable. This can be used with --append to both append and prepend at the same time.

In quotidian English, **append** means _to add_, and **prepend** means _to add at the beginning_. Clearly, these meanings don't apply to fish (or to computers in general?), for it's redundant to say that we can both _add x to y_ while also _adding x to the beginning of y._ If we can do the latter, then we can do the former. That is to say, by doing the latter, we do the former. If it's senseless to say that I am both running a race from the starting line and also running a race, then it's senseless to say that I am both prepending some data and also appending that data.

So, what do these terms mean to fish (or in the world of software development)? What are the implications for doing one over the other? And what does it mean to explicitly do both?

**Comment by bound-variable** [comment]
This question can be deleted. I know the answer now. And the question is useless since it arises from my own confusion.

**Comment by faho** [comment, marked as answer]
The simplest way to figure this out is to just try it:

```fish
> set foo middle
> set --append --prepend foo 1 2 3
> echo $foo
1 2 3 middle 1 2 3
```

So it simply takes the values it is supposed to add, adds them to the beginning and adds the same values, in the same order, to the end.

This is equivalent to

```fish
set foo 1 2 3 $foo 1 2 3
```

Is this super useful? Probably not. But there was no real reason to forbid it, it's consistent, so it's allowed.

I prefer to answer the question instead of letting it stay around or deleting it - https://xkcd.com/979/.
